src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/renderPaymentMethod.js

```diff
@@ -2,6 +2,8 @@ const store = require('../../../../store');
 const helpers = require('./helpers');
 const constants = require('../constants');
 
+const isSafari = /^((?!chrome|android|ios).)*safari/i.test(navigator.userAgent);
```

Comment:
we could avoid using `isSafari` flag here. We could use the `isAvailable` function from checkout.
we could call line 176 `handlePayment(options);` before line 165 and then:
`const node = store.componentsObj[paymentMethodID]?.node;`
 `if (node.isAvailable) { `
`const paymentMethodAvailable = await node.isAvailable()`
`if (!paymentMethodAvailable) {`
`delete store.componentsObj[paymentMethodID]`
`return`
`}`
`}`

Reply:
I added it in another place, I had to move the code around in new functions because of big complexity.
Please check